llvm/lib/Target/AArch64/Utils/AArch64SMEAttributes.cpp

```diff
@@ -65,6 +65,16 @@ SMEAttrs::SMEAttrs(const AttributeList &Attrs) {
 std::optional<bool>
 SMEAttrs::requiresSMChange(const SMEAttrs &Callee,
                            bool BodyOverridesInterface) const {
+  // If the caller has a streaming body and streaming compatible interface,
+  // we will have already conditionally enabled streaming mode on function
+  // entry. We need to disable streaming mode when a callee does not have A
+  // streaming interface, body, or streaming compatible interface.
+  if (hasStreamingBody() && hasStreamingCompatibleInterface())
+    return (!Callee.hasStreamingInterfaceOrBody() &&
+            !Callee.hasStreamingCompatibleInterface())
+               ? std::optional<bool>(false)
+               : std::nullopt;
+
```

Comment:
When I comment out this code, the tests still pass.

I'm also not sure what this change is supposed to do. Isn't this case already covered by:

```
  if (hasStreamingInterfaceOrBody() && Callee.hasStreamingInterface())
    return std::nullopt;
```
?

Reply:
The three states are to describe:
* `std::nullopt` = No change needed
* `std::optional<bool>(true)` = Change needed, smstart
* `std::optional<bool>(false)` = Change needed, smstop
I do agree the interface is confusing and could be better named.